### PR TITLE
{Attestation} `az attestation signer/policy`: Bump API version from 2022-10-01 to 2022-08-01

### PR DESCRIPTION
**Pull request by ZengTaoxu**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### General Guidelines

- [ ] Have you run `azdev style <YOUR_EXT>` locally? (`pip install azdev` required)
- [ ] Have you run `python scripts/ci/test_index.py -q` locally? (`pip install wheel==0.30.0` required)

For new extensions:

- [ ] My extension description/summary conforms to the [Extension Summary Guidelines](https://github.com/Azure/azure-cli/blob/dev/doc/extensions/extension_summary_guidelines.md).

### About Extension Publish

There is a pipeline to automatically build, upload and publish extension wheels.  
Once your pull request is merged into main branch, a new pull request will be created to update `src/index.json` automatically.  
You only need to update the version information in file setup.py and historical information in file HISTORY.rst in your PR but do not modify `src/index.json`.

### CODE REVIEW

**Comment by yonzhan** [comment]
Attestation

File: src/attestation/azext_attestation/aaz/latest/__init__.py [hunk 1]
```diff
@@ -1,13 +1,8 @@
...
-# Code generated by Microsoft (R) AutoRest Code Generator.
-# Changes may cause incorrect behavior and will be lost if the code is
-# regenerated.
-# --------------------------------------------------------------------------
-
-VERSION = "0.1.0"
+# Code generated by aaz-dev-tools
+# --------------------------------------------------------------------------------------------
 
+from ._clients import *
```
Comment (kairu-ms) [inline src/attestation/azext_attestation/aaz/latest/__init__.py:8]:
```suggestion
# pylint: skip-file
# flake8: noqa

from ._clients import *
```

File: src/attestation/azext_attestation/custom.py [hunk 2]
```diff
@@ -7,11 +7,498 @@
...
+
+
+def validate_provider_resource_id(self):
+    args = self.ctx.args
+    if has_value(args.id):
+        if has_value(args.resource_group) or has_value(args.name):
+            raise CLIError('Please omit --resource-group/-g or --name/-n if you have already specified --id.')
+        resource_id = args.id.to_serialized_data()
+        args.resource_group = resource_id.split('/')[4]
+        args.provider_name = resource_id.split('/')[8]
```
Comment (kairu-ms) [inline src/attestation/azext_attestation/custom.py:460]: It's better use `parse_resource_id` from azure.mgmt.core.tools

File: src/attestation/azext_attestation/custom.py [hunk 3]
```diff
@@ -7,11 +7,498 @@
...
+    elif not all([has_value(args.resource_group), has_value(args.name)]):
+        raise CLIError('Please specify both --resource-group/-g and --name/-n.')
```
Comment (kairu-ms) [inline src/attestation/azext_attestation/custom.py:462]: CLIError is deprecated

File: src/attestation/azext_attestation/custom.py [hunk 4]
```diff
@@ -7,11 +7,498 @@
...
+
+    def pre_operations(self):
+        validate_provider_resource_id(self)
+
+    def _output(self, *args, **kwargs):
+        args = self.ctx.args
+        show_args = {"resource_group": args.resource_group, "provider_name": args.provider_name, "attestation_type": args.attestation_type}
+        from azext_attestation.aaz.latest.attestation.policy import Show
+        token = Show(cli_ctx=self.cli_ctx)(command_args=show_args)['token']
+        return handle_policy_output(token)
```
Comment (kairu-ms) [inline src/attestation/azext_attestation/custom.py:496]: Same here

**Review by kairu-ms — APPROVED** [review]

**Pull request merged**